fs/_node_fs_file.ts

```diff
@@ -0,0 +1,211 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+import { getNodeFs, getNodeStream, getNodeTty, getNodeUtil } from "./_utils.ts";
+import { mapError } from "./_map_error.ts";
+import type { FileInfo, FsFile, SetRawOptions } from "./unstable_types.ts";
+import { toFileInfo } from "./_to_file_info.ts";
+
+/**
+ * The internal class to convert a Node file descriptor into a FsFile object.
+ */
+export class NodeFsFile implements FsFile {
+  #nodeFs = getNodeFs();
+  #nodeStream = getNodeStream();
+  #nodeTty = getNodeTty();
+  #nodeUtil = getNodeUtil();
+
+  #closed: boolean;
+  #rid: number;
+  #readableStream?: ReadableStream<Uint8Array>;
+  #writableStream?: WritableStream<Uint8Array>;
+
+  constructor(fd: number) {
+    this.#rid = fd;
+    this.#closed = false;
+  }
+
+  get readable(): ReadableStream<Uint8Array> {
+    if (this.#readableStream == null) {
+      const readStream = this.#nodeFs.createReadStream(null as unknown, {
+        fd: this.#rid,
+        autoClose: false,
+      });
+      this.#readableStream = this.#nodeStream.Readable.toWeb(readStream);
+    }
+    return this.#readableStream as ReadableStream;
+  }
+
+  get writable(): WritableStream<Uint8Array> {
+    if (this.#writableStream == null) {
+      const writeStream = this.#nodeFs.createWriteStream(null as unknown, {
+        fd: this.#rid,
+        autoClose: false,
+      });
+      this.#writableStream = this.#nodeStream.Writable.toWeb(writeStream);
+    }
+    return this.#writableStream as WritableStream;
+  }
+
+  [Symbol.dispose](): void {
+    if (!this.#closed) {
+      this.close();
+    }
+  }
+
+  close(): void {
+    this.#closed = true;
+    this.#nodeFs.closeSync(this.#rid);
+  }
+
+  isTerminal(): boolean {
+    return this.#nodeTty.isatty(this.#rid);
+  }
+
+  // deno-lint-ignore require-await no-unused-vars
+  async lock(exclusive?: boolean): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
+
+  // deno-lint-ignore no-unused-vars
+  lockSync(exclusive?: boolean): void {
+    throw new Error("Method not implemented.");
+  }
+
+  async read(p: Uint8Array): Promise<number | null> {
+    const nodeReadFd = this.#nodeUtil.promisify(this.#nodeFs.read);
```

Comment:
Can we store this function at `this.#nodeReadFs` in the constructor? It feels like too much overhead to call promisify for each read

Reply:
Or maybe store in this function only when it's not initialized yet.